cmd/gateway/s3x/gateway-s3x-object.go

```diff
@@ -176,42 +144,41 @@ func (x *xObjects) PutObject(
 			obinfo.ContentType = v
 		}
 	}
+	return obinfo
+}
+
+// PutObject creates a new object with the incoming data,
+func (x *xObjects) PutObject(
+	ctx context.Context,
+	bucket, object string,
+	r *minio.PutObjReader,
+	opts minio.ObjectOptions,
+) (minio.ObjectInfo, error) {
+	err := x.ledgerStore.AssertBucketExits(bucket)
+	if err != nil {
+		return minio.ObjectInfo{}, x.toMinioErr(err, bucket, "", "")
+	}
+
 	data, err := ioutil.ReadAll(r)
 	if err != nil {
-		return objInfo, x.toMinioErr(err, bucket, object, "")
+		return minio.ObjectInfo{}, x.toMinioErr(err, bucket, object, "")
 	}
-	obinfo.Size_ = int64(len(data))
-	dataHash, err := x.dagPut(ctx, data)
+	obinfo := newObjectInfo(bucket, object, len(data), opts)
+	dataHash, err := ipfsSaveBytes(ctx, x.dagClient, data)
```

Comment:
I made a mistake originally planning this. Object data itself needs to be stored as UnixFS data. Individual gRPC messages for TemporalX are capped at 4MB, so if the object data is larger than 4MB we will run into problems. Additionally storing object data as opposed to DAG's means the data can't be viewed on most gateways, whereas if we use UnixFS, the data can be viewed directly on IPFS gateways, etc... The `go-temporalx-client` package has helper `Upload` function that allows uploading a UnixFS file that's larger than 4MB without causing issues with gRPC max message sizes.

tl;dr - Any actual object data should now be stored as UnixFS on IPFS, and the MinIO protocol buffer objects (aka, the bucket buffer obj, and the object buffer obj) should continue to be stored as DAGs.

Reply:
This change has just been made, please check

Reply:
Hell yea, nice job.